tools/fido2-unprot.sh

```diff
@@ -15,37 +35,37 @@ do
     DEV_PATH=$(echo "$DEV_PATH_NAME" | cut -d' ' -f1)
     DEV_NAME=$(echo "$DEV_PATH_NAME" | cut -d' ' -f1 --complement)
     DEV_PRETTY=$(echo "$DEV_NAME (at '$DEV_PATH')")
-    if expr match "$(${PREFIX}fido2-token -I $DEV_PATH)" ".* credMgmt.* clientPin.*\|.* clientPin.* credMgmt.*" > /dev/null ; then
+    if expr match "$(${FIDO_TOOLS_PREFIX}fido2-token -I $DEV_PATH)" ".* credMgmt.* clientPin.*\|.* clientPin.* credMgmt.*" > /dev/null ; then
         printf "Enter PIN for $DEV_PRETTY once (ignore further prompts): "
         stty -echo
         read PIN
         stty echo
         printf "\n"
-        RESIDENT_RPS=$(echo "${PIN}\n" | setsid -w ${PREFIX}fido2-token -L -r $DEV_PATH | cut -d' ' -f3)
+        RESIDENT_RPS=$(echo "${PIN}\n" | setsid -w ${FIDO_TOOLS_PREFIX}fido2-token -L -r $DEV_PATH | cut -d' ' -f3)
         printf "\n"
         RESIDENT_RPS_COUNT=$(echo "$RESIDENT_RPS" | wc -l)
         FOUND=0
         for j in $(seq 1 $DEV_RESIDENT_RPS_COUNT)
         do
             RESIDENT_RP=$(echo "$RESIDENT_RPS" | sed "${j}q;d")
-            UNPROT_CREDS=$(echo "${PIN}\n" | setsid -w ${PREFIX}fido2-token -L -k $RESIDENT_RP $DEV_PATH | grep ' uvopt$' | cut -d' ' -f2,3,4)
+            UNPROT_CREDS=$(echo "${PIN}\n" | setsid -w ${FIDO_TOOLS_PREFIX}fido2-token -L -k $RESIDENT_RP $DEV_PATH | grep ' uvopt$' | cut -d' ' -f2,3,4)
             printf "\n"
             UNPROT_CREDS_COUNT=$(echo "$UNPROT_CREDS" | wc -l)
-            if test $UNPROT_CREDS_COUNT -gt 0 ; then
+            if [ $UNPROT_CREDS_COUNT -gt 0 ] ; then
                 FOUND=1
                 echo "Unprotected credentials on $DEV_PRETTY for '$RESIDENT_RP':"
                 echo "$UNPROT_CREDS"
             fi
         done
-        if test $FOUND -eq 0 ; then
+        if [ $FOUND -eq 0 ] ; then
             echo "No unprotected credentials on $DEV_PRETTY"
         fi
     else
         echo "$DEV_PRETTY cannot enumerate credentials"
         echo "Discovering unprotected SSH credentials only..."
         STUB_HASH=$(echo -n "" | sha256sum)
```

Comment:
With that I get: 
fido2-assert: fido_dev_get_assert: FIDO_ERR_INVALID_LENGTH

You probably want:
STUB_HASH=$(echo -n "" | openssl sha256 -binary | base64)

Then I get: 
fido2-assert: fido_dev_get_assert: FIDO_ERR_NO_CREDENTIALS

Reply:
I changed this to your recommendation. Both work for me on Linux, does this change mean that this now works for you on mac OS?

I am surprised that you are seeing the error message since stderr should have been redirected to `/dev/null`.

Reply:
It does not work yet on MacOS and would require several other changes for that.

Could it be that it worked for you because you never tested it with an unprotected SSH credential? 

FIDO_ERR_NO_CREDENTIALS and FIDO_ERR_INVALID_LENGTH exit codes both results in going through the else clause "No unprotected SSH credentials" by your script as it is now.

Reply:
Thanks for changing the commit message.

> Could it be that it worked for you because you never tested it with an unprotected SSH credential?
> 
> FIDO_ERR_NO_CREDENTIALS and FIDO_ERR_INVALID_LENGTH exit codes both results in going through the else clause "No unprotected SSH credentials" by your script as it is now.

I just tested this with a device without support for credential management and with no credentials, which should go through the correct branch. This is when `fido2-assert` should print `fido2-assert: fido_dev_get_assert: FIDO_ERR_NO_CREDENTIALS` to `stderr`, given that it fails to find a credential for the RP `ssh:`, but this is supressed by the redirect of `stderr` to `/dev/null`. It does print the message if I remove the `2> /dev/null` from `fido2-assert` in that case.